modules/distributor/distributor_ring.go

```diff
@@ -18,9 +18,10 @@ import (
 // is used to strip down the config to the minimum, and avoid confusion
 // to the user.
 type RingConfig struct {
-	KVStore          kv.Config     `yaml:"kvstore"`
-	HeartbeatPeriod  time.Duration `yaml:"heartbeat_period"`
-	HeartbeatTimeout time.Duration `yaml:"heartbeat_timeout"`
+	KVStore                kv.Config     `yaml:"kvstore"`
+	HeartbeatPeriod        time.Duration `yaml:"heartbeat_period"`
+	HeartbeatTimeout       time.Duration `yaml:"heartbeat_timeout"`
+	GracefulShutdownPeriod time.Duration `yaml:"graceful_shutdown_period"`
```

Comment:
we should put this on the main distributor config instead of the ring config

Reply:
I think that makes sense, went too fast on this one :) On it today

Reply:
It took time, but it is solved 😅